platform/platform-configuration-listener/src/test/java/org/codice/ddf/configuration/store/ConfigurationFileDirectoryTest.java

```diff
@@ -511,12 +572,70 @@ void moveFile(Path source, Path destination) throws IOException {
+        setUpDefaultDirectoryExpectations();
+        doThrow(new IOException()).when(configurationAdmin).listConfigurations(anyString());
+        ConfigurationFileDirectory configurationFileDirectory = createConfigurationFileDirectoryWithNoFiles();
+        configurationFileDirectory.export(exportedDirectoryPath);
+    }
+
+    @Test(expected = ConfigurationFileException.class)
+    public void testExportListConfigurationsInvalidSyntaxException()
+            throws IOException, InvalidSyntaxException, ConfigurationFileException {
+        setUpDefaultDirectoryExpectations();
+        doThrow(new InvalidSyntaxException("", "")).when(configurationAdmin)
+                .listConfigurations(anyString());
+        ConfigurationFileDirectory configurationFileDirectory = createConfigurationFileDirectoryWithNoFiles();
+        configurationFileDirectory.export(exportedDirectoryPath);
+    }
+
+    @Test(expected = ConfigurationFileException.class)
+    public void testExportConfigurationFileException()
+            throws IOException, InvalidSyntaxException, ConfigurationFileException {
+        setUpDefaultDirectoryExpectations();
+        when(configurationAdmin.listConfigurations(anyString()))
+                .thenReturn(new Configuration[] {configuration});
+        ConfigurationFileDirectory configurationFileDirectory = createConfigurationFileDirectoryWithNoFiles();
+        when(configurationFileFactory
+                .createConfigurationFile((Dictionary<String, Object>) anyObject()))
+                .thenThrow(new ConfigurationFileException(""));
+        configurationFileDirectory.export(exportedDirectoryPath);
+    }
+
+    @Test
+    public void testExportWhenNoConfigurations()
+            throws IOException, InvalidSyntaxException, ConfigurationFileException {
+        setUpDefaultDirectoryExpectations();
+        when(configurationAdmin.listConfigurations(anyString())).thenReturn(null);
+        ConfigurationFileDirectory configurationFileDirectory = createConfigurationFileDirectoryWithNoFiles();
+        configurationFileDirectory.export(exportedDirectoryPath);
+        verify(configurationFileFactory, never())
+                .createConfigurationFile((Dictionary<String, Object>) anyObject());
+    }
+
+    @Test
+    public void testExport()
+            throws IOException, InvalidSyntaxException, ConfigurationFileException {
+        setUpDefaultDirectoryExpectations();
+        when(configurationAdmin.listConfigurations(anyString()))
+                .thenReturn(new Configuration[] {configuration});
+        ConfigurationFileDirectory configurationFileDirectory = createConfigurationFileDirectoryWithNoFiles();
+        when(configurationFileFactory
+                .createConfigurationFile((Dictionary<String, Object>) anyObject()))
+                .thenReturn(configFile1);
+        configurationFileDirectory.export(exportedDirectoryPath);
+        verify(configFile1, atLeastOnce()).exportConfig(anyString());
+    }
+
```

Comment:
We should add `@NotNull` on the directory parameter of the `ConfigurationMigrationService.export()` and `ConfigurationFile.exportConfig()` methods, add the appropriate param checks and have unit tests for those scenarios.